src/containers/guest-home-page/registration-dialog/RegistrationDialog.tsx

```diff
@@ -0,0 +1,99 @@
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+
+import { FC, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { useModalContext } from '~/context/modal-context'
+import { UserRoleEnum } from '~/types'
+
+import { signup } from '~/constants'
+
+import RegistrationForm from '~/containers/guest-home-page/registration-form/RegistrationForm'
+import GoogleLogin from '../google-login/GoogleLogin'
+
+import studentImg from '~/assets/img/signup-dialog/student.svg'
+import tutorImg from '~/assets/img/signup-dialog/tutor.svg'
+
+import styles from '~/containers/guest-home-page/registration-dialog/RegistrationDialog.styles'
+
+interface RegistrationDialogProps {
+  defaultRole: UserRoleEnum
+}
+
+const RegistrationDialog: FC<RegistrationDialogProps> = ({ defaultRole }) => {
+  const { t } = useTranslation()
+  const { closeModal } = useModalContext()
+
+  // Minimal form state without validation
+  const [formData, setFormData] = useState({
+    email: '',
+    password: '',
+    role: defaultRole || 'Student',
```

Comment:
І тоді при ініціалізації formData передавати лише один параметр role
```suggestion
    role: defaultRole, 
```

Reply:
Згодна